Panic in TestCancelAndReadd

`TestCancelAndReadd` has a race condition that causes a panic due to:

1. A very short wait-to-start-time of `100 * time.Millisecond` which causes `wg.Done()` to be called too many times
2. Not using a fake clock such as the code in https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/util/queue/work_queue_test.go

```
=== RUN   TestCancelAndReadd
panic: sync: negative WaitGroup counter

goroutine 320 [running]:
panic(0x23a2620, 0xc420240440)
	/usr/local/go/src/runtime/panic.go:500 +0x1ae
sync.(*WaitGroup).Add(0xc420218230, 0xffffffffffffffff)
	/usr/local/go/src/sync/waitgroup.go:75 +0x27c
sync.(*WaitGroup).Done(0xc420218230)
	/usr/local/go/src/sync/waitgroup.go:100 +0x42
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/node.TestCancelAndReadd.func1(0xc4202bfa40, 0x1, 0xc400000020)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/node/timed_workers_test.go:118 +0x65
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/node.(*TimedWorkerQueue).getWrappedWorkerFunc.func1(0xc4202bfa40, 0x0, 0x0)
	github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/node/_test/_obj_test/timed_workers.go:90 +0xb6
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/node.CreateWorker.func1()
	github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/node/_test/_obj_test/timed_workers.go:49 +0x72
created by time.goFunc
	/usr/local/go/src/time/sleep.go:154 +0x78
```

cc @gmarek @davidopp @timothysc @liggitt @stevekuznetsov

xref: https://github.com/kubernetes/kubernetes/pull/41311

--- a/pkg/controller/node/timed_workers_test.go
+++ b/pkg/controller/node/timed_workers_test.go
@@ -89,7 +89,7 @@ func TestCancel(t *testing.T) {
 		return nil
 	})
 	now := time.Now()
-	then := now.Add(100 * time.Millisecond)
+	then := now.Add(time.Second)
 	queue.AddWork(NewWorkArgs("1", "1"), now, then)
 	queue.AddWork(NewWorkArgs("2", "2"), now, then)
 	queue.AddWork(NewWorkArgs("3", "3"), now, then)
@@ -119,7 +119,7 @@ func TestCancelAndReadd(t *testing.T) {
 		return nil
 	})
 	now := time.Now()
-	then := now.Add(100 * time.Millisecond)
+	then := now.Add(time.Second)
 	queue.AddWork(NewWorkArgs("1", "1"), now, then)
 	queue.AddWork(NewWorkArgs("2", "2"), now, then)
 	queue.AddWork(NewWorkArgs("3", "3"), now, then)

